### ISSUE TITLE
Error 500 on connect azure

### ISSUE THREAD

**Issue by rdrscan** [paolosalvatori/ServiceBusExplorer#87]
I'm trying to connect in the notification hub and the following error appears to me in the ServiceBusExplorer Log

> <18:04:17> MessagingFactory successfully created
> <18:04:49> Exception: O servidor remoto retornou um erro: (500) Erro Interno do Servidor. TrackingId:64f7003b-2adb-4ca7-976d-54b6daff50f8, SystemTracker:perseusmobile.servicebus.windows.net, Timestamp:24/03/2017 21:04:49
> <18:04:49> InnerException: O servidor remoto retornou um erro: (500) Erro Interno do Servidor.

Debugging on Fiddler I got the raw error message:

> HTTP/1.1 500 Internal Server Error
> Transfer-Encoding: chunked
> Content-Type: application/xml; charset=utf-8
> Server: Microsoft-HTTPAPI/2.0
> Date: Fri, 24 Mar 2017 21:03:01 GMT
> <Fault xmlns="http://schemas.microsoft.com/ws/2005/05/envelope/none"><Code><Value>Receiver</Value><Subcode><Value xmlns:a="http://schemas.microsoft.com/net/2005/12/windowscommunicationfoundation/dispatcher">a:InternalServiceFault</Value></Subcode></Code><Reason><Text xml:lang="en-US">The server was unable to process the request due to an internal error.  For more information about the error, either turn on IncludeExceptionDetailInFaults (either from ServiceBehaviorAttribute or from the &lt;serviceDebug&gt; configuration behavior) on the server in order to send the exception information back to the client, or turn on tracing as per the Microsoft .NET Framework SDK documentation and inspect the server trace logs.</Text></Reason></Fault>

I verified the access key and it is the correct FullSharedAccessSignature with all permissions. I tried the last ServiceBusExplorer 4.0.57, and some other versions 4.0.56, 4.0.55, etc...

I have other application that also uses WindowsAzure.ServiceBus v4.0.0 and Microsoft.Azure.NotificationHubs v1.0.7 and got the error on fiddler: HTTP/1.1 401 40103: Invalid authorization token signature

Is anyone with this problem? 
Tks

**Comment by paolosalvatori** [comment]
Hi @rdrscan could you please send me a repro that does not use my tool? So I can forward it to the product group. Thanks!
Ciao
Paolo

**Comment by rdrscan** [comment]
Hello @paolosalvatori !
I made a mistake testing the nuget package Microsoft.Azure.NotificationHubs. Now I call tell you it is working on a simple console application (other methods like GetAllRegistrationsAsync is also working): _NotificationHubClient.CreateClientFromConnectionString(connectionString, hubPath)_

But the error 500 persists in ServiceBusExplorer with the same connection string.
I realized that the SharedAccessKeyName changed. I had this working for at least 2 years:

Endpoint=sb://myservice.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=MyScretKeyHere

And today I get a connection string from azure and the SharedAccessKeyName is not the same. Now is **DefaultFullSharedAccessSignature**:
Endpoint=sb://myservice.servicebus.windows.net/;SharedAccessKeyName=DefaultFullSharedAccessSignature;SharedAccessKey=MyScretKeyHere

Maybe azure has changed the default SharedAccessKeyName. I didn't understand very well.
